src/analyses/constant_propagator.cpp

```diff
@@ -380,58 +516,75 @@ Function: constant_propagator_domaint::valuest::merge
 
 bool constant_propagator_domaint::valuest::merge(const valuest &src)
 {
-  //nothing to do
+  // dummy
+  const symbol_tablet symbol_table;
+  const namespacet ns(symbol_table);
+
+  // nothing to do
   if(src.is_bottom)
     return false;
 
-  //just copy
+  // just copy
   if(is_bottom)
   {
-    replace_const = src.replace_const;
-    is_bottom = src.is_bottom;
+    assert(!src.is_bottom);
+    replace_const=src.replace_const; // copy
+    is_bottom=false;
     return true;
   }
 
-  bool changed = false;
+  assert(!is_bottom && !src.is_bottom);
```

Comment:
I'd like to keep this one. It's rather trivial but it forms a precondition for the code block below. That helps me as I change the code or move things around.